### ISSUE TITLE
Add macOS native libs

### ISSUE THREAD

**Issue by TraceLD** [kazuyaujihara/build-rdkit#9]
Hi,
would it be possible for you to add the option to build macOS native RDKFuncs?

**Comment by k-ujihara** [comment]
I believe it can do it with small modification. but I do not have recent Mac. 
Could you build it?

**Comment by TraceLD** [comment]
Yep, I managed to build them before you commented and put them up on my nuget (as I wasn't sure when you would reply and needed them urgently). 

Link: https://www.nuget.org/packages/TraceLD.RDKit.CSharp
You can pull out the binary itself from the nuget file itself as nuget packages are just archives so you can open it in 7zip or some other archive software (it's in runtimes/macos-arm64). I've tested it on M1 MacBook Air with Monterey. I should also be able to ask a friend who has an Intel Mac to compile it for x64 macOS Monterey.

**Comment by TraceLD** [comment]
I also updated the P/Invoke code that loads the binary to support macOS, it's on my GitHub, although it only supports new .NET as I used some code that isn't available in Standard for determining the OS (hence why my package targets NET6.0)

**Comment by k-ujihara** [comment]
Clever work. I believe that binary (shared object) itself should be work both .NET Core 5.0 and 6.0. I will check it later.

**Comment by TraceLD** [comment]
> Clever work. I believe that binary (shared object) itself should be work both .NET Core 5.0 and 6.0. I will check it later.

Yeah, the binary should work fine on anything .NET Core/.NET. Just my C# dll is for NET 6.0 since I added code that was added in I believe .NET Core 3 and .NET 6.0 is the current LTS version so I thought it made sense to target that.

**Comment by k-ujihara** [comment]
Now the target Framework is .NET Standard 2.0 because this library does not uses new APIs and also work with .NET Framework. In fact, Microsoft recommended target is .NET Standard 2.0. 
https://docs.microsoft.com/en-us/dotnet/standard/net-standard
